hasura/migrations/1644261710606_create_search_guild/up.sql

```diff
@@ -0,0 +1,14 @@
+CREATE EXTENSION pg_trgm;
+CREATE INDEX guildname_gin_idx ON guild
+USING GIN (guildname gin_trgm_ops);
+
+CREATE OR REPLACE FUNCTION public.search_guilds(search text)
```

Comment:
You have again created a custom SQL function to do an `ilike` search which is handled already by the existing GraphQL endpoint. Rather than creating custom code, please create a GraphQL query for the data you wish to retrieve.

You query, `searchGuilds`, is likely to look alot like [getGuild](https://github.com/MetaFam/TheGame/blob/develop/packages/web/graphql/queries/guild.ts#L20-L25) except with an `_ilike` rather than an `_eq` and in the function you'll append and wildcards to the search string to get it to match more broadly: e.g.: *(I'm writing this freehand, so there may be errors)*

```javascript
const guildSearch = /* GraphQL */ `
  query SearchGuilds($search: String!) {
    guild(where: { guildname: { _ilike: $search } }) {
      ...GuildFragment
    }
  }
  ${GuildFragment}
`;

export const searchGuilds = async (
  search: string = '',
): Promise<Array<GuildFragmentType> | undefined> => {
  const { data, error } = (
    await client
    .query<SearchGuildsQuery, SearchGuildsQueryVariables>(
      guildSearch, { search: `%${search}%` }
    )
    .toPromise()
  );

  if(!data && error) throw new Error(error)

  return data?.guild;
};

Reply:
Recall that after you add a new GraphQL query, you have to run `yarn generate` to create the types.

If you get an error that `SearchGuildsQuery` or `SearchGuildsQueryVariables` isn't defined, that is likely the issue.

Reply:
yeah sorry forgot to remove the guild SQL function . I only removed the player's one